### PR TITLE
Add `_ring_iso` helper to help avoid some `Base.getproperty` methods

### PR DESCRIPTION
**Pull request by fingolfin**
I plan to get rid of the rest, too, but that'll take some more time.

### CODE REVIEW

File: src/Groups/matrices/MatGrp.jl [hunk 1]
```diff
@@ -198,28 +198,32 @@ function assign_from_description(G::MatrixGroup)
...
+    if T === QQBarFieldElem
+      # get all matrix entries into one vector
+      entries = reduce(vcat, vec(collect(matrix(g))) for g in gens(G))
+      # construct a number field over which all matrices are already defined
+      nf, nf_to_QQBar = number_field(QQ, entries)
+      psi = pseudo_inv(nf_to_QQBar)
+      iso = iso_oscar_gap(nf)
+      G.ring_iso = MapFromFunc(G.ring, codomain(iso),
+                               x -> iso(preimage(nf_to_QQBar, x)),
+                               y -> preimage(iso,nf_to_QQBar(y))
```
Comment (joschmitt) [inline src/Groups/matrices/MatGrp.jl]: I know that this was there before, but shouldn't it be
```suggestion
                               y -> nf_to_QQBar(preimage(iso, y))
```
Also the map `psi` seems not to be used in this function.

**Review by ThomasBreuer — APPROVED** [review]
Yes, thanks, good idea.

File: src/Groups/matrices/MatGrp.jl [hunk 2]
```diff
@@ -198,28 +198,32 @@ function assign_from_description(G::MatrixGroup)
...
 end
 
+function _ring_iso(G::MatrixGroup{T}) where T
+  if !isdefined(G, :ring_iso)
+    if T === QQBarFieldElem
+      # get all matrix entries into one vector
+      entries = reduce(vcat, vec(collect(matrix(g))) for g in gens(G))
+      # construct a number field over which all matrices are already defined
+      nf, nf_to_QQBar = number_field(QQ, entries)
+      psi = pseudo_inv(nf_to_QQBar)
```
Comment (fingolfin) [inline src/Groups/matrices/MatGrp.jl]:
```suggestion
```

File: src/Groups/matrices/MatGrp.jl (hunk 1 above)
Comment (fingolfin) [reply src/Groups/matrices/MatGrp.jl]: Ah indeed!

`psi` was from an earlier iteration of this code and I forgot to delete it.

I guess we should have a test case for this problem, too...

**Pull request merged**